### ISSUE TITLE
nvim unresponsive after 4 keystrokes

### ISSUE THREAD

**Issue by pierreganty** [neovim/neovim#12293]
- `nvim --version`:
NVIM v0.4.3
Build type: Release

- `vim -u DEFAULTS` (version: ) behaves differently?
yes, vim 8.0 does not stall

- Operating system/version:
macOS 10.14.6

- Terminal name/version:
Apple Terminal
- `$TERM`:
xterm-256color

### Steps to reproduce using `nvim -u NORC`

i
CTRL-K then ^ then backspace

### Actual behaviour

nvim gets stuck foreever

### Expected behaviour

nvim does not get stuck

**Comment by pierreganty** [comment]
Using the output of Activity Monitor on the nvim process I obtain

```
Call graph:
    2736 Thread_7649768   DispatchQueue_1: com.apple.main-thread  (serial)
    + 2736 start  (in libdyld.dylib) + 1  [0x7fff60aff3d5]
    +   2736 main  (in nvim) + 10367  [0x10980e8f9]
    +     2736 state_enter  (in nvim) + 196  [0x1098cf6c8]
    +       2736 normal_execute.llvm.15691625082451191132  (in nvim) + 4448  [0x10983a497]
    +         2736 invoke_edit  (in nvim) + 79  [0x109840e71]
    +           2736 edit  (in nvim) + 2081  [0x10975a9d9]
    +             2736 state_enter  (in nvim) + 196  [0x1098cf6c8]
    +               2736 insert_handle_key  (in nvim) + 7973  [0x109767307]
    +                 2736 do_insert_char_pre  (in nvim) + 115  [0x109764d56]
    +                   2736 ins_apply_autocmds  (in nvim) + 52  [0x109767b1a]
    +                     2736 apply_autocmds_group.llvm.16698147497802074250  (in nvim) + 1646  [0x1097e7374]
    +                       2736 do_cmdline  (in nvim) + 2556  [0x1097b95b1]
    +                         2736 do_one_cmd  (in nvim) + 10900  [0x1097bcfd4]
    +                           2736 ex_call  (in nvim) + 577  [0x10976f19a]
    +                             2736 get_func_tv  (in nvim) + 473  [0x10976fc2d]
    +                               2736 call_func  (in nvim) + 3702  [0x10976d66d]
    +                                 2736 do_cmdline  (in nvim) + 2556  [0x1097b95b1]
    +                                   2736 do_one_cmd  (in nvim) + 10900  [0x1097bcfd4]
    +                                     2736 script_host_execute  (in nvim) + 175  [0x1097b327d]
    +                                       2736 eval_call_provider  (in nvim) + 405  [0x10977a276]
    +                                         2736 call_func  (in nvim) + 3702  [0x10976d66d]
    +                                           2736 do_cmdline  (in nvim) + 2556  [0x1097b95b1]
    +                                             2736 do_one_cmd  (in nvim) + 10900  [0x1097bcfd4]
    +                                               2736 ex_return  (in nvim) + 111  [0x109778aa7]
    +                                                 2736 eval0  (in nvim) + 75  [0x10976be02]
    +                                                   2736 eval1.llvm.17164366973993554196  (in nvim) + 50  [0x10976c041]
    +                                                     2736 eval3  (in nvim) + 48  [0x10977be7a]
    +                                                       2736 eval4  (in nvim) + 47  [0x10977bfe9]
    +                                                         2736 eval5  (in nvim) + 53  [0x10977c791]
    +                                                           2736 eval6  (in nvim) + 49  [0x10977cbf3]
    +                                                             2736 eval7  (in nvim) + 4450  [0x10977e00c]
    +                                                               2736 get_func_tv  (in nvim) + 473  [0x10976fc2d]
    +                                                                 2736 call_func  (in nvim) + 1011  [0x10976cbea]
    +                                                                   2736 f_call  (in nvim) + 413  [0x109786938]
    +                                                                     2736 call_func  (in nvim) + 1011  [0x10976cbea]
    +                                                                       2736 f_rpcrequest  (in nvim) + 694  [0x10978bfda]
    +                                                                         2736 rpc_send_call  (in nvim) + 431  [0x109830e2f]
    +                                                                           2736 loop_poll_events  (in nvim) + 138  [0x1097a316a]
    +                                                                             2736 uv_run  (in libuv.1.dylib) + 360  [0x109a19fb3]
    +                                                                               2736 uv__io_poll  (in libuv.1.dylib) + 872  [0x109a29c6b]
    +                                                                                 2736 kevent  (in libsystem_kernel.dylib) + 10  [0x7fff60c3a78a]
```

**Comment by erw7** [comment]
@pierreganty In your Call graph, `f_rpcrequest()` is called. That seems very strange to me. I started with `nvim -u NORC`, attached it to `gdb`, set a breakpoint to `f_rpcrequest` and did your procedure operation, but it doesn't hit the breakpoint. To me, it looks like you are using some kind of plugin. If that's the case, you first need to find out which plugin is causing the problem.

**Comment by pierreganty** [comment]
With `nvim -u NONE` I cannot reproduce the problem anymore.
What's a good way to find the plugin that's responsible?

**Comment by glacambre** [comment]
> What's a good way to find the plugin that's responsible?

I'd say disabling half of your plugins, seeing if you can reproduce or 
not, disabling the remaining half, seeing if you can reproduce… Until 
you find the culprit.

**Comment by pierreganty** [comment]
That's right. [ultisnips](https://github.com/SirVer/ultisnips) seems at fault here. I'll go post an issue there.

**Comment by pierreganty** [comment]
SirVer/ultisnips#1227

**Comment by pierreganty** [comment]
It turns out the problem was actually in pynvim. It is fixed by neovim/pynvim/pull/439 .
